### ISSUE TITLE
[R] Handle "unmatched" argument in joins

### ISSUE THREAD

**Issue by nealrichardson** [apache/arrow#33988]
### Describe the enhancement requested

New feature in dplyr 1.1: https://www.tidyverse.org/blog/2023/01/dplyr-1-1-0-joins/#unmatched-rows

I don't know that we can support this, but we should at least be aware of the argument and raise a helpful message if someone tries to use it.

### Component(s)

R

**Comment by westonpace** [comment]
I suspect it wouldn't be too hard to add support for this in C++ if someone wanted to.  Well, I will clarify, "would not be a large code change".  It will probably require some learning curve for anyone not familiar with the asof join node.

**Comment by nealrichardson** [comment]
I don't think this is only for the asof join, I think it has meaning for most joins

**Comment by westonpace** [comment]
Oops, yes, I meant "hash join node".  I think unmatched would not make sense in an asof join case ("closest" somewhat implies you will be throwing rows away).

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 2 years. Please remove the stale label or comment below, or this issue will be closed in 14 days.
